Lib/test/test_monitoring.py

```diff
@@ -2083,6 +2083,30 @@ def callback(code, instruction_offset):
         sys.monitoring.restart_events()
         sys.monitoring.set_events(0, 0)
 
+    @test.support.requires_working_socket()  # For asyncio
+    def test_yield_async_generator(self):
+        # gh-129013: Async generators have a special type that they
+        # use to yield values. This type shouldn't be exposed by PY_YIELD
+        # events.
+        asyncio = test.support.import_helper.import_module("asyncio")
+
+        async def gen():
+            yield 42
+
+        async def main():
+            async for _ in gen():
+                pass
+
+        def handle_yield(code, offset, value):
+            self.assertEqual(value, 42)
+
+        sys.monitoring.use_tool_id(0, "test")
+        sys.monitoring.register_callback(0, sys.monitoring.events.PY_YIELD, handle_yield)
+        sys.monitoring.set_events(0, sys.monitoring.events.PY_YIELD)
+
+        asyncio.run(main())
+        sys.monitoring.set_events(0, 0)
```

Comment:
This would seem slightly cleaner.
```suggestion
        try:
            asyncio.run(main())
        finally:
             sys.monitoring.clear_tool_id(0)
```

Reply:
Not disagreeing, but I'd rather stay consistent with the rest of the file. `try`/`finally` doesn't seem to be used with `set_events(0, 0)` in these tests, so I'm going to skip it for now. Sounds reasonable to add that in a follow-up PR, though.